ports/esp32/sdkconfig.h

```diff
@@ -1,6 +1,6 @@
 #define CONFIG_TRACEMEM_RESERVE_DRAM 0x0
 #define CONFIG_BT_RESERVE_DRAM 0x0
-#define CONFIG_ULP_COPROC_RESERVE_MEM 0
+#define CONFIG_ULP_COPROC_RESERVE_MEM 4096
```

Comment:
what does this practically mean? what are the valid limits to this value?

Reply:
There's 8kB of RTC SLOW memory in the ESP32, and the lower half of it is available to users. These 4kB are divided between space for variables placed into .rtc.data/.rtc.rodata by the linker (via RTC_DATA_ATTR/RTC_RODATA_ATTR attributes), and the area reserved for the ULP.

So practically you need to set this to 4096 minus the size of RTC_DATA_ATTR/RTC_RODATA_ATTR variables allocated by micropython.